packages/editor/src/components/DiagramPanel.tsx

```diff
@@ -18,11 +18,108 @@ import {
   diagramState,
   fileContentsSelector,
   ProgramFile,
+  RogerState,
   WorkspaceMetadata,
   workspaceMetadataSelector,
 } from "../state/atoms";
 import BlueButton from "./BlueButton";
 
+/**
+ * Fetch url, but try local storage first using a name.
+ * Update local storage if the file is fetched via url
+ *
+ * @param name The short name of the file to fetch
+ * @param url The url to fetch, if not found locally
+ * @returns Promise that resolves to the fetched string or undefined if the fetch failed
+ */
+const fetchResource = async (
```

Comment:
From the diff it looks like this and `pathResolver` just moved from below to up here; any other changes?